landing-page/content/common/index-and-statistics-format.md

```diff
@@ -0,0 +1,129 @@
+---
+url: index-and-statistics-format
+toc: false
+---
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Index and statistics file format
+
+This is a specification for the Plain Format for Iceberg Statistics, a file
+format designed to store information such as statistics about data managed in an
+Iceberg table that cannot be stored directly within the Iceberg manifest. A
+statistics file contains arbitrary pieces of information (here called "blobs"),
+along with metadata necessary to interpret them. The blobs supported by Iceberg
+are documented at [Blob types](#blob-types).
+
+## Format specification
+
+A file conforming to the format specification should have the structure as
+described below.
+
+### File structure
+
+The file has the following structure
+
+```
+Magic Blob₁ Blob₂ ... Blobₙ Footer
+```
+
+where
+
+- `Magic` is four bytes 0x50, 0x46, 0x49, 0x53 (short for: Plain Format for
+  Indices and Statistics),
+- `Blobᵢ` is i-th blob contained in the file, to be interpreted by application
+  according to the footer,
+- `Footer` is defined below.
+
+### Footer structure
+
+Footer has the following structure
+
+```
+Magic FooterPayload FooterPayloadSize Reserved Flags FileFormatVersion Magic
+```
+
+where
+
+- `Magic` four bytes, same as at the beginning of the file.
+- `FooterPayload` optionally compressed, UTF-8 encoded JSON payload describing the
+  blobs in the file, with the structure described below,
+- `FooterPayloadSize` is a length in bytes of the `FooterPayload` (compressed),
+  stored as 4 byte integer,
+- `Reserved` is 4 bytes reserved for future use, currently should be written as
+  0x00, 0x00, 0x00, 0x00,
+- `Flags` a 4 byte integer for boolean flags
+  - 0 (lowest bit): whether `FooterPayload` is compressed
+  - all other bits are reserved for future use and should be set to 0 on write
+- `FileFormatVersion` is a number, stored as 4 byte integer.
```

Comment:
Can this be part of the "flags" area rather than a 4-byte integer? 4 bytes seems unnecessary considering we are unlikely to have even 255 versions of this spec.

Reply:
Also, why not put this in the JSON payload?

Reply:
> Also, why not put this in the JSON payload?

file format implies layout of the fixed-width section.
if we move file version inside JSON we won't ever be able to change the Footer layout

> Can this be part of the "flags" area rather than a 4-byte integer? 4 bytes seems unnecessary considering we are unlikely to have even 255 versions of this spec.

I don't think we will have that many versions, but i also don't think squeezing those 3 bytes out of this section helps anyone. Do you feel strongly about that it shouldn't be an integer?

Reply:
Every field that isn't in the payload adds complexity and an opportunity for someone to mess up implementation. For binary sections like this, I think simpler is much better.

Rather than adding this to the payload, which would limit flexibility, I think what I would do is combine this with magic bytes. Both Avro and Parquet do that, using `Obj1` and `PAR1` respectively.

Reply:
Sure, i can combine this with magic bytes. This will remove the FileFormatVersion field.

@alexjo2144 @losipiuk please ack